fmp_data/lc/__init__.py

```
@@ -83,36 +68,129 @@ def validate_api_keys(
+                    f"Registered {len(endpoints_for_batch)} endpoints from {group_name}"
                 )
-
-        if endpoints_dict:
-            registry.register_batch(endpoints_dict)
+            except Exception as e:
+                logger.error(f"Failed to register {group_name} endpoints: {e}")
 
     return registry
 
 
+def create_vector_store(
```

Pylint (3.3.7)

[refactor] Too many arguments (7/5)

(R0913)